pkg/admission/attributes.go

```diff
@@ -21,14 +21,16 @@ import (
 )
 
 type attributesRecord struct {
+	kind      string
 	namespace string
 	resource  string
 	operation string
 	object    runtime.Object
 }
 
-func NewAttributesRecord(object runtime.Object, namespace, resource, operation string) Attributes {
+func NewAttributesRecord(object runtime.Object, kind, namespace, resource, operation string) Attributes {
```

Comment:
I think I can make an argument to provide Resource, Kind, and APIVersion in the long run.

Since the AttributesRecord is only constructed in the 4 places, I don't think the extra argument is a major concern.  The Attributes interface makes the consumer side easy enough I think.